src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ConstructorSymbolReferenceFinder.cs

```diff
@@ -30,28 +31,65 @@ protected override bool CanFind(IMethodSymbol symbol)
                 _ => false,
             };
 
+        protected override Task<ImmutableArray<string>> DetermineGlobalAliasesAsync(IMethodSymbol symbol, Project project, CancellationToken cancellationToken)
+        {
+            var containingType = symbol.ContainingType;
+            return GetAllMatchingGlobalAliasNamesAsync(project, containingType.Name, containingType.Arity, cancellationToken);
+        }
+
         protected override async Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
             IMethodSymbol symbol,
+            HashSet<string>? globalAliases,
             Project project,
             IImmutableSet<Document>? documents,
             FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
+            var containingType = symbol.ContainingType;
             var typeName = symbol.ContainingType.Name;
 
+            using var _ = ArrayBuilder<Document>.GetInstance(out var result);
+
+            await AddDocumentsAsync(
+                project, documents, typeName, result, cancellationToken).ConfigureAwait(false);
+
+            if (globalAliases != null)
+            {
+                foreach (var globalAlias in globalAliases)
+                {
+                    await AddDocumentsAsync(
+                        project, documents, globalAlias, result, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            result.AddRange(await FindDocumentsAsync(
+                project, documents, containingType.SpecialType.ToPredefinedType(), cancellationToken).ConfigureAwait(false));
+
+            result.AddRange(await FindDocumentsWithGlobalAttributesAsync(
+                project, documents, cancellationToken).ConfigureAwait(false));
+
+            result.AddRange(symbol.MethodKind == MethodKind.Constructor
+                ? await FindDocumentsWithImplicitObjectCreationExpressionAsync(project, documents, cancellationToken).ConfigureAwait(false)
+                : ImmutableArray<Document>.Empty);
+
+            return result.ToImmutable();
+        }
+
+        private static async Task AddDocumentsAsync(
+            Project project,
+            IImmutableSet<Document>? documents,
+            string typeName,
+            ArrayBuilder<Document> result,
+            CancellationToken cancellationToken)
+        {
             var documentsWithName = await FindDocumentsAsync(project, documents, cancellationToken, typeName).ConfigureAwait(false);
-            var documentsWithType = await FindDocumentsAsync(project, documents, symbol.ContainingType.SpecialType.ToPredefinedType(), cancellationToken).ConfigureAwait(false);
```

Comment:
this got inlined into the caller.  we don't want to search for built-in types twice, we only want to do the multi-passes for real names and aliases.